beater/config/rum.go

```diff
@@ -140,20 +140,37 @@ func replaceVersion(pattern, version string) string {
 	return regexObserverVersion.ReplaceAllLiteralString(pattern, version)
 }
 
+func (s *SourceMapping) Unpack(inp *common.Config) error {
+	cfg := tmpSourceMapping(*defaultSourcemapping())
+	if err := inp.Unpack(&cfg); err != nil {
+		return errors.Errorf("error unpacking sourcemapping config: %w", err)
+	}
+	*s = SourceMapping(cfg)
+	if inp.HasField("elasticsearch") {
+		s.esConfigured = true
+	}
+	return nil
+}
+
+type tmpSourceMapping SourceMapping
```

Comment:
Perhaps move this inside the Unpack method, and comment it to make it clear that it exists just to remove the custom "Unpack" method?